### ISSUE TITLE
Code style enum initialization

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet/ddnet#4626]
There are two ways used to initialize flag enums.

Can we pick one and enforce it?

https://github.com/ddnet/ddnet/blob/390e54f627400642aa55d98106fea2f4a342f12b/src/engine/textrender.h#L11-L23

A probably not very fair attempt of measure the current usage

```
$ grep -rA2 " = 1 << 0," | grep -B2 " = 1 << 1" | wc -l
42
$ grep -rA2 " = 2," | grep -B2 " = 4" | wc -l
81
```

```
$ grep -r '^[[:space:]]*[A-Z_]*FLAG[A-Z_]*[[:space:]]*=[[:space:]]*[0-9]* <<' | wc -l
45
$ grep -r '^[[:space:]]*[A-Z_]*FLAG[A-Z_]*[[:space:]]*=[[:space:]]*[0-9]*' | grep -v '<<' | wc -l
52
```

**Comment by def-** [comment]
Just switch them all to enum class

**Comment by edg-l** [comment]
the second one has redundancy in its name, if its used it should be

```cpp
 enum ETextAlignment 
 { 
 	LEFT = 1 << 0, 
 	CENTER = 1 << 1, 
 	RIGHT = 1 << 2, 
 }; 
```

**Comment by Chairn** [comment]
I prefer second one for full name as its more clear, but value assignment is shorter and clearer with the first form.

**Comment by Robyt3** [comment]
See https://github.com/ddnet/ddnet/issues/9992#issuecomment-2770357337. I'd say those shouldn't be enums, but I prefer the style `1 << 0`, `1 << 1` etc. for the initialization regardless.

**Comment by ChillerDragon** [comment]
@Robyt3 okay so for existing code ``1 << 0`` is now officially recommended. I would still like to fully wrap this up. Close all related issues and add it to CONTRIBUTING.md. I am okay with going with `static constexpr uint32_t` for flags (https://github.com/ddnet/ddnet/issues/9992#issuecomment-2770357337)

**Comment by Robyt3** [comment]
> [@Robyt3](https://github.com/Robyt3) okay so for existing code `1 << 0` is now officially recommended. I would still like to fully wrap this up. Close all related issues and add it to CONTRIBUTING.md. I am okay with going with `static constexpr uint32_t` for flags ([#9992 (comment)](https://github.com/ddnet/ddnet/issues/9992#issuecomment-2770357337))

Yes, I would appreciate if we could close #4626 (this), #5524, #9606 and #9992 by summarizing the conclusions of the discussions in CONTRIBUTING.md.
